### PR TITLE
openshift-e2e-test: Error if no TEST_SKIPS match

### PR DESCRIPTION
**Pull request by pierreprinetti**
This change prints an error message to STDOUT when the passed
environment variable TEST_SKIPS does not match any test.

Previously, the call to `grep` would just fail and the container would
fail after printing "Skipping tests:".

### CODE REVIEW

**Comment by pierreprinetti** [comment]
Example: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/20577/rehearse-20577-pull-ci-openshift-installer-release-4.9-e2e-openstack-byon/1419744493061541888

File: ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh [hunk 1]
```diff
@@ -215,7 +215,7 @@ function suite() {
         TESTS="$(openshift-tests run --dry-run --provider "${TEST_PROVIDER}" "${TEST_SUITE}")"
         echo "${TESTS}" | grep -v "${TEST_SKIPS}" >/tmp/tests
         echo "Skipping tests:"
-        echo "${TESTS}" | grep "${TEST_SKIPS}"
+        echo "${TESTS}" | grep "${TEST_SKIPS}" || { exit_code=$?; echo 'Error: no tests were found matching the TEST_SKIPS regex:'; echo "$TEST_SKIPS"; return $exit_code }
```
Comment (vrutkovs) [inline ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh:218]:
```
/bin/bash: -c: line 221: syntax error near unexpected token `fi'
```
in all jobs

**Review by vrutkovs — CHANGES_REQUESTED** [review]

**Review by vrutkovs — APPROVED** [review]
```
Skipping tests:
"[sig-cli] Kubectl Port forwarding With a server listening on 0.0.0.0 should support forwarding over websockets [Suite:openshift/conformance/parallel] [Suite:k8s]"
"[sig-cli] Kubectl Port forwarding With a server listening on localhost should support forwarding over websockets [Suite:openshift/conformance/parallel] [Suite:k8s]"
"[sig-cli] Kubectl client Simple pod should support exec through an HTTP proxy [Suite:openshift/conformance/parallel] [Suite:k8s]"
"[sig-cli] Kubectl client Simple pod should support exec through kubectl proxy [Suite:openshift/conformance/parallel] [Suite:k8s]"
```
in `ci/rehearse/openshift/installer/release-4.9/e2e-openstack-byon` log, no weird failures

File: ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh (hunk 1 above)
Comment (pierreprinetti) [reply ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh]: oh no :(

I have added the missing semicolon and the tests now look better.

Thanks for the /approve!

**Pull request merged**